common/lib/dependabot/clients/azure.rb

```diff
@@ -190,6 +190,33 @@ def create_pull_request(pr_name, source_branch, target_branch,
           "/pullrequests?api-version=5.0", content.to_json)
       end
 
+      # rubocop:disable Metrics/ParameterLists
+      def autocomplete_pull_request(pull_request_id, auto_complete_set_by, merge_commit_message,
+                                    delete_source_branch = true, squash_merge = true, merge_strategy = "squash",
+                                    trans_work_items = true)
+
+        content = {
+          autoCompleteSetBy: {
+            id: auto_complete_set_by
+          },
+          completionOptions: {
+            mergeCommitMessage: merge_commit_message,
+            deleteSourceBranch: delete_source_branch,
+            squashMerge: squash_merge,
+            mergeStrategy: merge_strategy,
+            transitionWorkItems: trans_work_items,
+            autoCompleteIgnoreConfigIds: []
+          }
+        }
+
+        response = patch(source.api_endpoint +
+                           source.organization + "/" + source.project +
+                           "/_apis/git/repositories/" + source.unscoped_repo +
+                           "/pullrequests/" + pull_request_id.to_s + "?api-version=6.0", content.to_json)
```

Comment:
In the 5.0 there is `squashMerge` as boolean, from 5.1 it has string value. With which is designed this PR. I agree with politeness since it does not create technical dept in a new code.

Reply:
I downgraded API version to 5.1